pkg/services/ppmshipment/rules_test.go

```
@@ -357,123 +357,100 @@ func (suite *PPMShipmentSuite) TestValidationRules() {
+					oldPPMShipment: defaultOldShipmentValues,
+					newPPMShipment: models.PPMShipment{
+						ShipmentID:         id,
+						EstimatedIncentive: &estimatedIncentive,
+						AdvanceRequested:   truePointer,
+						Advance:            &highAdvance,
+					},
+				},
+				"advance less than 1": {
+					oldPPMShipment: defaultOldShipmentValues,
+					newPPMShipment: models.PPMShipment{
+						ShipmentID:         id,
+						EstimatedIncentive: &estimatedIncentive,
+						AdvanceRequested:   truePointer,
+						Advance:            &lessThanOneAdvance,
+					},
+				},
+				"advance is not nil": {
+					oldPPMShipment: defaultOldShipmentValues,
+					newPPMShipment: models.PPMShipment{
+						ShipmentID:         id,
+						EstimatedIncentive: &estimatedIncentive,
+						AdvanceRequested:   nil,
+						Advance:            &normalAdvance,
+					},
+				},
+				"advance requested is true while advance is nil": {
+					oldPPMShipment: defaultOldShipmentValues,
+					newPPMShipment: models.PPMShipment{
+						ShipmentID:         id,
+						EstimatedIncentive: &estimatedIncentive,
+						AdvanceRequested:   truePointer,
+						Advance:            nil,
+					},
+				},
 			}
 
-			newPPMShipment := models.PPMShipment{
-				ShipmentID:         shipmentID,
-				EstimatedIncentive: &estimatedIncentive,
-				AdvanceRequested:   nil,
-				Advance:            &newAdvance,
+			for name, testCase := range testCases {
+				suite.Run(name, func() {
+					err := checkAdvance().Validate(suite.AppContextForTest(), testCase.newPPMShipment, &testCase.oldPPMShipment, nil)
+					switch verr := err.(type) {
+					case *validate.Errors:
+						suite.True(verr.HasAny())
+						suite.Contains(verr.Keys(), "advance")
+					default:
+						suite.Failf("expected *validate.Errors", "%t - %v", err, err)
+					}
+				})
 			}
-
-			err := checkAdvance().Validate(suite.AppContextForTest(), newPPMShipment, &oldPPMShipment, nil)
-			suite.Error(err)
```

Tests were refactored to reduce duplicate code and because `suite.Error(err)` doesn't actually catch the scenarios when checkAdvance doesn't add an error message. For example, in our [old test](https://github.com/transcom/mymove/blob/master/pkg/services/ppmshipment/rules_test.go#L433-L455) we are providing an advance amount of 1 cent and checking if it is < 1 cent. This should pass the advance check  without adding errors and should throw an error since we are expecting a failure but receiving an error with no validation errors .